Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -111,6 +113,18 @@ open class ChatChannelVC: _ViewController,
             }
             self?.setChannelControllerToComposerIfNeeded(cid: self?.channelController.cid)
             self?.messageComposerVC.updateContent()
+
+            let pagination = self?.channelController.channelQuery.pagination?.parameter
+            switch pagination {
+            case let .around(messageId):
+                guard let message = self?.channelController.dataStore.message(id: messageId) else {
+                    return
+                }
+
+                self?.messageListVC.jumpToMessage(message, withHighlight: true)
```

Comment:
Shouldn't we skip synchronize altogether if we are directly jumping to a message?

Reply:
Right now the synchronize loads the mid-page, and then the jumpToMessage will jump to the selected message. (It won't load the messages twice, since the message will already be available at this time)